packages/core/src/shared/ui/sam/stackPrompter.ts

```diff
@@ -0,0 +1,51 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { StackSummary } from 'aws-sdk/clients/cloudformation'
+import { getAwsConsoleUrl } from '../../awsConsole'
+import { DefaultCloudFormationClient } from '../../clients/cloudFormationClient'
+import { samSyncUrl } from '../../constants'
+import { createCommonButtons } from '../buttons'
+import { createQuickPick } from '../pickerPrompter'
+import * as nls from 'vscode-nls'
+import { getRecentResponse } from '../../sam/utils'
+
+export const localize = nls.loadMessageBundle()
+
+const canPickStack = (s: StackSummary) => s.StackStatus.endsWith('_COMPLETE')
+const canShowStack = (s: StackSummary) =>
+    (s.StackStatus.endsWith('_COMPLETE') || s.StackStatus.endsWith('_IN_PROGRESS')) && !s.StackStatus.includes('DELETE')
+
+export function createStackPrompter(client: DefaultCloudFormationClient, mementoRootKey: string) {
+    const recentStack = getRecentResponse(mementoRootKey, client.regionCode, 'stackName')
+    const consoleUrl = getAwsConsoleUrl('cloudformation', client.regionCode)
+    const items = client.listAllStacks().map((stacks) =>
+        stacks.filter(canShowStack).map((s) => ({
+            label: s.StackName,
+            data: s.StackName,
+            invalidSelection: !canPickStack(s),
+            recentlyUsed: s.StackName === recentStack,
+            description: !canPickStack(s) ? 'stack create/update already in progress' : undefined,
+        }))
+    )
+
+    return createQuickPick(items, {
+        title: 'Select a CloudFormation Stack',
+        placeholder: 'Select a stack (or enter a name to create one)',
+        filterBoxInputSettings: {
+            label: 'Create a New Stack',
+            transform: (v) => v,
+        },
+        buttons: createCommonButtons(samSyncUrl, consoleUrl),
```

Comment:
If this is intended to be used by both sync/deploy, why this only add syncUrl in the help button?

Reply:
I think this refactoring really bring up many questions about the current logic. 

In master branch, the `deploy.ts` import `createStackPrompter()` directly from `sync.ts`. I believe this was intended, or at least okay to share (!?!?!?).

![Screenshot 2024-11-14 at 15 24 38](https://github.com/user-attachments/assets/5f90a447-6a17-4948-b9a8-6998cebf625e)

Reply:
Then we should fix this bug along with this chagne

Reply:
Need to update the deployUrl.